docs/design/specs/runtime-async.md

```diff
@@ -29,24 +29,31 @@ Unlike sync methods, async methods support suspension. Suspension allows async m
 
 Async methods also do not have matching return type conventions as sync methods. For sync methods, the stack should contain a value convertible to the stated return type before the `ret` instruction. For async methods, the stack should be empty in the case of `Task` or `ValueTask`, or the type argument in the case of `Task<T>` or `ValueTask<T>`.
 
-Async methods support the following suspension points:
+Async methods support suspension using one of the following methods:
 
-* Calling an `Await` method on `Task/ValueTask/Task<T>/ValueTask<T>` as defined below. If the callee suspends, the caller will suspend as well.
-* Using new .NET runtime APIs to "await" an "INotifyCompletion" type. The signatures of these methods shall be:
   ```C#
   namespace System.Runtime.CompilerServices
   {
       public static class RuntimeHelpers
       {
           [MethodImpl(MethodImplOptions.Async)]
-          public static Task AwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : INotifyCompletion { ... }
+          public static void AwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : INotifyCompletion { ... }
           [MethodImpl(MethodImplOptions.Async)]
-          public static Task UnsafeAwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : ICriticalNotifyCompletion
+          public static void UnsafeAwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter) where TAwaiter : ICriticalNotifyCompletion
+          
+          [MethodImpl(MethodImplOptions.Async)]
+          public static void Await(Task task);
+          [MethodImpl(MethodImplOptions.Async)]
+          public static void Await(ValueTask task);
+          [MethodImpl(MethodImplOptions.Async)]
+          public static T Await<T>(Task<T> task);
+          [MethodImpl(MethodImplOptions.Async)]
+          public static T Await<T>(ValueTask<T> task);
       }
   }
   ```
 
-Each of the above methods will have semantics analogous to the current AsyncTaskMethodBuilder.AwaitOnCompleted/AwaitUnsafeOnCompleted methods. After calling this method, it can be presumed that the task has completed. These methods are only legal to call from inside async methods.
+These methods are only legal to call inside async methods. The `...AwaitAwaiter...` methods will have semantics analogous to the current `AsyncTaskMethodBuilder.AwaitOnCompleted/AwaitUnsafeOnCompleted` methods. After calling either method, it can be presumed that the task or awaiter has completed. The `Await` methods perform suspension like the `AwaitAwaiter...` methods, but are optimized for calling on the return value of a call to an async method. To achieve maximum performance, the IL sequence of two `call` instructions -- one to the async method and immediately one to the `Await` method -- should be preferred.
```

Comment:
> These methods are only legal to call inside async methods

From a language perspective, we'd probably have to make an actual change to C# enforce this as a compiler error. We could do a warning without any issue though.